tests/diff/patch.c

```diff
@@ -610,3 +610,47 @@ void test_diff_patch__line_counts_with_eofnl(void)
 
 	git_buf_free(&content);
 }
+
+void test_diff_patch__can_strip_bad_utf8(void)
```

Comment:
All that we want to do in this test is to just make sure that the diff of two files produces the correct result. So in fact we don't really need to add the complete test repository here, which would be a bit wasteful. Instead, you could just use `git_diff_buffers`.

Reply:
I could add the two files (https://raw.githubusercontent.com/stanhu/rugged-encoding-test/3c4f7762042ad735d97d4781c4c7ee7d4486646c/Japanese.md) and (https://raw.githubusercontent.com/stanhu/rugged-encoding-test/387914a182ae0868364ec09fadb1beb47690fd4a/Japanese.md) in as character buffers, but wouldn't this pollute the test even more since we have to escape UTF-8 characters)? 

Ideally I'd be able to simplify the test case so that it boils down to a few lines, but when I tried earlier it was challenging to reproduce. I can try again.

Reply:
@pks-t Ok, I think I figured out a better test case. I just mirrored this one: https://github.com/git/git/blob/master/t/t4025-hunk-header.sh

Reply:
Oh, these tests are great! Thanks a lot!